### ISSUE TITLE
Celeste crashing after starting for a few frames

### ISSUE THREAD

**Issue by Ellaquin** [pixelomer/Celeste-ARM64#3]
The game loads for a few frames, but then crashes. I am on Ubuntu Bionic 18.04, and running this on my switch.
[errorLog.txt](https://github.com/pixelomer/Celeste-ARM64/files/9689545/errorLog.txt)

**Comment by pixelomer** [comment]
Seems like `LD_LIBRARY_PATH` isn't working for some reason. For now you could try copying the files in `lib-arm64` into the Celeste directory and that should fix the issue.

    cp lib-arm64/* .

**Comment by skydark719** [comment]
That's strange, I tried running those commands, and the problem it's still there, with the exact same error log.

**Comment by pixelomer** [comment]
Seems like there was a bug in the code... Could you try again with this build?  
[Celeste-ARM64-prebuilt.zip](https://github.com/pixelomer/Celeste-ARM64/files/10300435/Celeste-ARM64-prebuilt.zip)

Relevant change:
```diff
diff --git a/fmod/fmod_preload.c b/fmod/fmod_preload.c
index 5bfbddc..c0b962d 100644
--- a/fmod/fmod_preload.c
+++ b/fmod/fmod_preload.c
@@ -49,11 +49,11 @@ void *dlopen(const char *filename, int flags) {
                strcpy(filename_copy, filename);
                char *name = basename(filename_copy);
                if (strncmp(name, "libfmodstudio.so.10", 19) == 0) {
-                       filename = "libfmodstudio.so.13";
+                       filename = "libfmodstudio.so.13.6";
                        target_pt = &fmodstudio;
                }
                else if (strncmp(name, "libfmod.so.10", 13) == 0) {
-                       filename = "libfmod.so.13";
+                       filename = "libfmod.so.13.6";
                        target_pt = &fmod;
                }
                void *result = dlopen_real(filename, flags);
```

**Comment by skydark719** [comment]
Even with that, I patched it again with the game, then I transfer the files to the Switch, and the error is appearing another time:
[errorLog.txt](https://github.com/pixelomer/Celeste-ARM64/files/10300470/errorLog.txt)

**Comment by skydark719** [comment]
Okay, now it works, I saw the same issue on Discord and fixes it.
ln -s libfmodstudio.so.13.6 libfmodstudio.so.13
ln -s libfmod.so.13.6 libfmod.so.13

Executing both commands on lib-arm64 folder.

**Comment by xplodingcreeper2** [comment]
Great, thank you, I will try it out and post the results here

**Comment by canihavemy30** [comment]
I had a similar problem and i tried some of the stuff mentioned here and one worked - now I'm only getting this error.

Ver 1.4.0.0
1/3/2023 5:12:34 PM
System.DllNotFoundException: libfmodstudio.so.10 assembly:<unknown assembly> type:<unknown type> member:(null)
  at (wrapper managed-to-native) FMOD.Studio.System.FMOD_Studio_System_Create(intptr&,uint)
  at FMOD.Studio.System.create (FMOD.Studio.System& studiosystem) [0x00005] in <d7b2abb405ec4c97954549f0a02e4a41>:0 
  at Celeste.Audio.Init () [0x00010] in <d7b2abb405ec4c97954549f0a02e4a41>:0 
  at Celeste.GameLoader.LoadThread () [0x0000c] in <d7b2abb405ec4c97954549f0a02e4a41>:0 
  at Celeste.RunThread.RunThreadWithLogging (System.Action method) [0x00000] in <d7b2abb405ec4c97954549f0a02e4a41>:0

**Comment by pixelomer** [comment]
Fixed in [2023.02.04](https://github.com/pixelomer/Celeste-ARM64/releases/2023.02.04).
